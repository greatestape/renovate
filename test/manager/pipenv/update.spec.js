const fs = require('fs');
const { updateDependency } = require('../../../lib/manager/pipenv/update');

const pipfile = fs.readFileSync('test/_fixtures/pipenv/Pipfile1', 'utf8');

describe('manager/pipenv/update', () => {
  describe('updateDependency', () => {
    it('replaces existing value', () => {
      const upgrade = {
        depName: 'some-package',
        newValue: '==1.0.1',
        depType: 'packages',
      };
      const res = updateDependency(pipfile, upgrade);
      expect(res).not.toEqual(pipfile);
      expect(res.includes(upgrade.newValue)).toBe(true);
      expect(res).toMatchSnapshot();
    });
    it('upgrades dev packages', () => {
      const upgrade = {
        depName: 'dev-package',
        newValue: '==0.2.0',
        depType: 'dev-packages',
      };
      const res = updateDependency(pipfile, upgrade);
      expect(res).not.toEqual(pipfile);
      expect(res.includes(upgrade.newValue)).toBe(true);
      expect(res).toMatchSnapshot();
    });
    it('returns null if error', () => {
      const res = updateDependency(null, null);
      expect(res).toBe(null);
    });
  });
});
