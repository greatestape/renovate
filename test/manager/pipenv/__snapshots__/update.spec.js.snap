// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager/pipenv/update updateDependency replaces existing value 1`] = `
"[[source]]
url = \\"https://pypi.org/simple\\"
verify_ssl = true
name = \\"pypi\\"

[[source]]
url = \\"http://example.com/private-pypi/\\"
verify_ssl = false
name = \\"private-pypi\\"

[packages]
some-package = \\"==1.0.1\\"
some-other-package = \\"==1.0.0\\"
_invalid-package = \\"==1.0.0\\"
invalid-version = \\"==0 0\\"

[dev-packages]

[requires]
python_version = \\"3.6\\"
"
`;
